src/opensles/AudioStreamOpenSLES.cpp

```diff
@@ -326,7 +357,7 @@ Result AudioStreamOpenSLES::close_l() {
 SLresult AudioStreamOpenSLES::enqueueCallbackBuffer(SLAndroidSimpleBufferQueueItf bq) {
     SLresult result = (*bq)->Enqueue(
             bq, mCallbackBuffer[mCallbackBufferIndex].get(), mBytesPerCallback);
-    mCallbackBufferIndex = (mCallbackBufferIndex + 1) % kBufferQueueLength;
+    mCallbackBufferIndex = (mCallbackBufferIndex + 1) % kBufferQueueLengthMax;
```

Comment:
Should this use mBufferQueueLength in any way? This seems to put data into unused buffers if mBufferQueueLength is small

Reply:
This code cycles through allocated buffers. If we do then we should also only allocate mBufferQueueLength. 
The reality of how all this works is not what one would expect. We do not actually queue multiple buffers into the linked list because the list is in the sender thread and basically useless. The "length" is only used as an indirect signal to AudioFlinger to tell it how large a FIFO to allocate.

Reply:
I used mBufferQueueLength to save memory.